lua/clangd_extensions/switch_source_header.lua

```diff
@@ -14,10 +17,14 @@ local function handler(_err, uri)
     }, {})
 end
 
-return {
-    switch_source_header = function()
-        vim.lsp.buf_request(0, "textDocument/switchSourceHeader", {
-            uri = vim.uri_from_bufnr(0),
-        }, handler)
-    end,
-}
+local M = {}
+
+function M.switch_source_header()
+    local bufnr = curr_buf()
+    local client = vim.lsp.get_clients({ bufnr = bufnr })[1]
```

Comment:
people might have all sorts of clients attached to buffers which also have clangd attached, i don't think we can just take the first one. it's now how the function is implemented in neovim either: https://github.com/neovim/neovim/blob/78f4994627b7d9c3b85f30028fe55ff38024a39d/runtime/lua/vim/lsp.lua#L1182

Reply:
I was afraid of that. Perhaps filtering both by `bufnr` and `id` (or any other useful filters) might be a solution?